content/300-guides/400-migrate-to-prisma/02-migrate-from-sequelize.mdx

```diff
@@ -0,0 +1,1217 @@
+
+
+#### `/user/:userId/profile`
+
+The `/user/:userId/profile` handler is currently implemented as follows:
+
+```js file=src/controllers/user.js
+const setUserBio = async (req, res) => {
+  const { userId } = req.params
+  const { bio } = req.body
+
+  try {
+    const user = await User.findOne({
+      where: {
+        id: Number(userId),
+      },
+    })
+
+    const updatedUser = await user.createProfile({ bio })
+
+    return res.json(updatedUser)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the route is implemented as follows:
+
+```js file=src/controllers/user.js
+const setUserBio = async (req, res) => {
+  const { userId } = req.params
+  const { bio } = req.body
+
+  try {
+    const user = await prisma.user.update({
+      where: { id: Number(userId) },
+      data: {
+        profile: {
+          create: { bio },
+        },
+      },
+    })
+
+    return res.json(user)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+### 4.3. Replacing queries in `PUT` requests
+
+The REST API has one route that accept a `PUT` request:
+
+- `/addPostToCategory?postId=POST_ID&categoryId=CATEGORY_ID`: Adds the post with `POST_ID` to the category with `CATEGORY_ID`
+
+Let's dive into the route handlers that implement these requests.
+
+#### `/addPostToCategory?postId=POST_ID&categoryId=CATEGORY_ID`
+
+The `/addPostToCategory?postId=POST_ID&categoryId=CATEGORY_ID` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const addPostToCategory = async (req, res) => {
+  const { postId, categoryId } = req.query
+
+  try {
+    const post = await Post.findOne({
+      where: { id: postId },
+    })
+
+    const category = await Category.findOne({
+      where: { id: categoryId },
+    })
+
+    const updatedPost = await post.addCategory(category)
+
+    return res.json(updatedPost)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the route is implemented as follows:
+
+```js file=src/controllers/post.js
+const addPostToCategory = async (req, res) => {
+  const { postId, categoryId } = req.query
+
+  try {
+    const post = await prisma.post.update({
+      data: {
+        PostToCategories: {
+          create: {
+            Categories: {
+              connect: { id: Number(categoryId) },
+            },
+          },
+        },
+      },
+      where: {
+        id: Number(postId),
+      },
+    })
+
+    return res.json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+Note that this Prisma Client can be made less verbose by modeling the relation as an [implicit many-to-many relation](#implicit-many-to-many-relations) instead. In that case, the query would look as follows:
+
+```js file=src/controllers/posts.js
+const post = await prisma.post.update({
+  data: {
+    category: {
+      connect: { id: categoryId },
+    },
+  },
+  where: { id: postId },
+});
+```
+
+## More
```

Comment:
I love that you added a couple of new sections with specific behaviour of Sequelize here 😍

Reply:
Thank you 😊